src/common/Gateway.sol

```diff
@@ -187,7 +187,9 @@ contract Gateway is Auth, Recoverable, IGateway {
 
     /// @inheritdoc IGateway
     function addUnpaidMessage(uint16 centrifugeId, bytes memory message) external auth {
-        _addUnpaidBatch(centrifugeId, message, gasService.messageGasLimit(centrifugeId, message));
+        uint128 gasLimit = gasService.messageGasLimit(centrifugeId, message) + extraGasLimit;
+        extraGasLimit = 0;
+        _addUnpaidBatch(centrifugeId, message, gasLimit);
```

Comment:
This was a bug. The unpaid message should take into account any extra gas added when it was created